### PR TITLE
ALFREDAPI-470: Search limit & memory problems

### PR DESCRIPTION
**Pull request by WimCrols**
Fixes https://xenitsupport.jira.com/browse/ALFREDAPI-470

- [x] Is [CHANGELOG.md](https://github.com/xenit-eu/alfred-api/blob/master/CHANGELOG.md) extended?
- [ ] Does this PR avoid breaking the API? 
    Breaking changes include adding, changing or removing endpoints and/or JSON objects used in requests and responses.
- [ ] Does the PR comply to REST HTTP result codes policy outlined in the [user guide](https://docs.xenit.eu/alfred-api/stable-user/rest-api/index.html#rest-http-result-codes)?
- [ ] Is error handling done through a method annotated with `@ExceptionHandler` in the webscript classes?
- [x] Does the PR follow our [coding styleguide and other active procedures](https://xenitsupport.jira.com/wiki/spaces/XEN/pages/624558081/XeniT+Enhancement+Proposals+XEP)?
- [ ] Is usage of `this.` prefix avoided?

See [README.md](./README.md) for full explanation.

### CODE REVIEW

File: apix-impl/src/main/java/eu/xenit/apix/alfresco/search/SearchService.java [hunk 1]
```diff
@@ -150,17 +152,16 @@ protected SearchParameters buildSearchParameters(SearchQuery postQuery) {
         return searchParameters;
     }
 
-    // These 2 methods are overridden in the 4.2 impl
     protected void setSearchLimit(SearchParameters searchParameters, int max) {
```
Comment (RVanhuysseXenit) [inline apix-impl/src/main/java/eu/xenit/apix/alfresco/search/SearchService.java:155]: To improve clarity, can we change the getter and setter to getMaxItems and setMaxItems?

File: apix-impl/src/test/java/eu/xenit/apix/tests/search/SearchServiceUnitTest.java [hunk 2]
```diff
@@ -121,6 +122,27 @@ public void testOrderBy_withMultivalueProperty_throwsIllegalArgumentException()
         apixSearchServiceMocked.query(query);
     }
 
+    @Test
+    public void testLimitDefault() {
```
Comment (RVanhuysseXenit) [inline apix-impl/src/test/java/eu/xenit/apix/tests/search/SearchServiceUnitTest.java:126]: Same rename here

**Review by RVanhuysseXenit — CHANGES_REQUESTED** [review]

File: apix-impl/src/test/java/eu/xenit/apix/tests/search/SearchServiceUnitTest.java (hunk 2 above)
Comment (WimCrols) [reply apix-impl/src/test/java/eu/xenit/apix/tests/search/SearchServiceUnitTest.java:126]: The parameter is actually called `limit` so I think the name should stay.

**Comment by kerkhofsd** [comment]
If I'm not mistaking, we still don't have a test that makes sure the query to / response from Solr is limited? 

Given the amount of hassle this can cause, I think it is feasible to investigate if it's possible to add such a test? 
(When I investigated the issue, I added this: https://github.com/xenit-eu/alfred-api/commit/3fa1c32a29d68f4cf9dddabae1f11e7e24c3e5d1, what might or might not be a source of inspiration.)

**Review by RVanhuysseXenit — DISMISSED** [review]

**Comment by WimCrols** [comment]
> If I'm not mistaking, we still don't have a test that makes sure the query to / response from Solr is limited?
> 
> Given the amount of hassle this can cause, I think it is feasible to investigate if it's possible to add such a test?
> (When I investigated the issue, I added this: [3fa1c32](https://github.com/xenit-eu/alfred-api/commit/3fa1c32a29d68f4cf9dddabae1f11e7e24c3e5d1), what might or might not be a source of inspiration.)

@kerkhofsd Correct, because that would be a test for Solr, not for Alfred API in my opinion.

**Review by RVanhuysseXenit — APPROVED** [review]

**Pull request merged**